src/pages/keyword-analyzer/KeywordAnalyzer.vue

```diff
@@ -0,0 +1,595 @@
+<script setup>
+import { ref, computed, onMounted, watch } from 'vue'
+import { MaInput, MaButton, MaSelect, MaPagination, MaProgress, MaSelect2, MaToggle } from "@mobileaction/action-kit"
+
+// Props to receive static text from parent
+const props = defineProps({
+  staticText: {
+    type: String,
+    default: 'This is a sample text for keyword analysis. You can analyze this text to find keywords and their frequency. The analysis will show you how often each keyword appears in the text and what percentage of the total words it represents.'
+  }
+})
+
+// Reactive data
+const inputText = ref('')
+const selectedKeywords = ref([])
+const analysisResults = ref([])
+const isAnalyzing = ref(false)
+const analysisMode = ref('specific') // 'specific' or 'all'
+
+// Pagination data
+const currentPage = ref(1)
+const perPage = ref(10)
+
+// Analysis mode options
+const analysisModeOptions = [
+  {
+    value: 'specific',
+    label: 'Choose your keywords'
+  },
+  {
+    value: 'all',
+    label: 'All keywords with density'
+  }
+]
+
+// Keyword options (will be populated from text analysis)
+const keywordOptions = ref([])
+
+// Computed properties
+const hasResults = computed(() => analysisResults.value.length > 0)
+const totalWords = computed(() => {
+  if (!inputText.value.trim()) return 0
+  return inputText.value.toLowerCase().match(/\b\w+\b/g)?.length || 0
+})
+
+// Pagination computed properties
+const totalItems = computed(() => analysisResults.value.length)
+const totalPages = computed(() => Math.ceil(totalItems.value / perPage.value))
```

Comment:
```suggestion
// Pagination computed properties
const totalItems = computed(() => analysisResults.value.length)
const totalPages = computed(() => Math.ceil(totalItems.value / perPage.value))
```
Do we need these? Seems like there is an overwork for pagination and seemed too AI ish

Reply:
Yes since without these, the pagination functionality dysfunctions with each change in the items per page